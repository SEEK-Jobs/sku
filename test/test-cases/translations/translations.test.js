/* eslint-disable jest/expect-expect */
const path = require('path');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const startAssetServer = require('../../utils/assetServer');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');

describe('translations', () => {
  let closeAssetServer;

  beforeAll(async () => {
    await runSkuScriptInDir('build', appDir);
    closeAssetServer = await startAssetServer(4758, distDir);
  });

  afterAll(() => {
    closeAssetServer();
  });

  it('should render en', async () => {
    const app = await getAppSnapshot('http://localhost:4758/en');
    expect(app).toMatchSnapshot();
  });

  it('should render fr', async () => {
    const app = await getAppSnapshot('http://localhost:4758/fr');
    expect(app).toMatchSnapshot();
  });
});
