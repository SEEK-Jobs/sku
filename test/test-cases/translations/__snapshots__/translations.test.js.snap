// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translations should render en 1`] = `
SCRIPTS: Array [
  "/runtime-c23b2036f7ddf87040b9.js",
  "/vendors~main-3a565b7aae951dc0a9f6.js",
  "/main-f9b4000e73189f5b46fc.js",
  "/en-translations-e581123a4492c402af5e.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="en-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
      </div>
    </div>
    <script id="__SKU_CLIENT_CONTEXT__"
            type="application/json"
    >
      {"language":"en"}
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      [0]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="en-translations"
            src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`translations should render fr 1`] = `
SCRIPTS: Array [
  "/runtime-c23b2036f7ddf87040b9.js",
  "/vendors~main-3a565b7aae951dc0a9f6.js",
  "/main-f9b4000e73189f5b46fc.js",
  "/fr-translations-f4898ff95f2574ccea80.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="fr-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Bonjour sku
      </div>
    </div>
    <script id="__SKU_CLIENT_CONTEXT__"
            type="application/json"
    >
      {"language":"fr"}
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      [1]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
    >
      {"namedChunks":["fr-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="fr-translations"
            src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`translations should support query parameters 1`] = `
SCRIPTS: Array [
  "/runtime-c23b2036f7ddf87040b9.js",
  "/vendors~main-3a565b7aae951dc0a9f6.js",
  "/main-f9b4000e73189f5b46fc.js",
  "/en-translations-e581123a4492c402af5e.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
    <link data-chunk="en-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
      </div>
    </div>
    <script id="__SKU_CLIENT_CONTEXT__"
            type="application/json"
    >
      {"language":"en"}
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      [0]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
    <script async
            data-chunk="en-translations"
            src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
